index.html

```diff
@@ -555,7 +556,102 @@ <h2>Dictionary {{RTCRtpEncodingParameters}} Members</h2>
+              {{RTCRtpEncodingParameters/codec}} is marked as a feature at risk
+              due to lack of support from implementers.
+            </p>
+          </div>
+      </dd>
     </dl>
+    <h2>Modifications to existing procedures</h2>
+    <h3>setParameters()</h3>
+    <p>Add the following steps in the parameter validation for {{RTCRtpSender/setParameters()}},
+      before any validation of <var>scalabilityMode</var> from [[WEBRTC-SVC]]:</p>
+    <ol>
+      <li><p>Let <var>codecs</var> be <var> parameters</var>.{{RTCRtpParameters/codecs}}.</p>
+      </li>
+      <li>
+        <p>If <var>codecs</var> is not an empty list:</p>
+        <ol>
+          <li>
+            <p>If any <var>codec</var> parameter in <var>encodings</var> does not match
+              any entry in <var>codecs</var>, return a promise [= rejected =] with a newly
+              [= exception/created =] {{InvalidModificationError}}.</p>
+          </li>
+        </ol>
+      </li>
+      <li>
+        <p>Else (<var>codecs</var> is an empty list), if <var>transceiver</var>.
+          {{RTCRtpTransceiver/[[PreferredCodecs]]}} is not an empty list:</p>
+        <ol>
+          <li>
+            <p>If any <var>codec</var> parameter in <var>encodings</var> does not match
+              any entry in the <var>transceiver</var>.{{RTCRtpTransceiver/[[PreferredCodecs]]}},
+              return a promise [= rejected =] with a newly [= exception/created =]
+              {{InvalidModificationError}}.</p>
+          </li>
+        </ol>
+      </li>
+      <li>
+        <p>Else ({{RTCRtpTransceiver/[[PreferredCodecs]]}} is an empty list):</p>
+        <ol>
+          <li>
+            <p>If any <var>codec</var> parameter in <var>encodings</var> does not match
+              any codec in {{RTCRtpSender.getCapabilities(kind)}}.<code>codecs</code>,
+              return a promise [= rejected =] with a newly [= exception/created =]
+              {{InvalidModificationError}}.</p></li>
+        </ol>
+      </li>
+      <li>
+        <p>If the user agent does not support setting the codec for any encoding or mixing
+          different codec values on the different encodings, return a promise [= rejected =]
+          with a newly [= exception/created =] {{OperationError}}.
+        </p>
+      </li>
+    </ol>
+    <h3>addTransceiver()</h3>
+    <p>Add the following steps in the <var>sendEncodings</var> validation for
+      {{RTCPeerConnection/addTransceiver()}}, before any validation of
+      <var>scalabilityMode</var> from [[WEBRTC-SVC]]:</p>
+    <ol>
+      <li>
+        <p>If any <var>codec</var> parameter in <var>sendEncodings</var> does not match any codec in
+        {{RTCRtpSender.getCapabilities(kind)}}.<code>codecs</code>,
+        [= exception/throw =] an {{OperationError}}.</p>
+      </li>
+      <li>
+        <p>If the user agent does not support changing the codec for any encoding or mixing
+          different codec values on the different encodings, 
```

Comment:
"mixing different codec values" seems a bit vague. What exactly are the requirements for not throwing OperationError?

Reply:
`OperationError` is already part of the specification when `setParameters()` cannot allocate resources. Chrome is able to throw those when unknown errors occur deep in the pipeline, which we associate with a resource allocation issue for now.
This describes a new behaviour though and I believe returning an `OperationError` in case of a complex unsupported input might be relevant.